### ISSUE TITLE
bug: Disabled button still clickable on Safari

### ISSUE THREAD

**Issue by josh-byster** [ionic-team/ionic-framework#23865]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

Hi, it seems like for `IonButton` elements with little text (e.g. 1 character), setting the `disabled` property to true after it has been clicked does not prevent subsequent `onClick` events.
For example, in my minimal reproducible example, I have 2 buttons—one with more text than the other. If I click either button, they change color, as I expect (I set `disabled` to be true after one press).
However, I can keep clicking the second button and the `onClick` event keeps firing. This doesn't seem to happen when I click the first button after it becomes disabled.

![image](https://user-images.githubusercontent.com/8664074/131779282-1b2f4ff4-4bf1-48c0-a858-9cb8a103204d.png)

![image](https://user-images.githubusercontent.com/8664074/131779072-3f55e284-878e-4a33-8069-6ea79ba36d25.png)

![image](https://user-images.githubusercontent.com/8664074/131779053-0b371cb2-3ea2-4e45-a3b4-14c01fc84dae.png)

### Expected Behavior

I expect that any time the user clicks on a disabled button, the `onClick` event doesn't fire.

### Steps to Reproduce

Open in Safari, as this does not seem to happen in Chrome. I'm using Version 14.1.1 (16611.2.7.1.4). This also happens on iOS 14.17.1 Safari.

1. Click on "Hello World" or "2".
2. You will see a popup that a button was pressed. Both buttons will now become disabled.
3. Click "2". 
4. You will see a popup that the button was pressed. This should not happen.
5. Click "Hello World". 
6. On my machine, nothing pops up, though I can't say why this is.

### Code Reproduction URL

https://github.com/josh-byster/ionic-button-disable-bug-report/blob/main/src/App.tsx

### Ionic Info

```
Ionic:

   Ionic CLI       : 5.4.16 (/usr/local/lib/node_modules/ionic)
   Ionic Framework : @ionic/react 5.7.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v14.17.1 (/usr/local/bin/node)
   npm    : 6.14.13
   OS     : macOS Big Sur
```

### Additional Information

This seemed to work fine in the last few months (I don't know for sure), but seemed to just start happening very recently. I could be wrong, and we may have just noticed this now, but I haven't upgrade any dependencies in my app since January, and this just seemed to start happening on our deployed version (was running ionic `5.5.2`).

This also doesn't seem to be a specific issue with `width`, since I set `width` to a large value and still saw the same results—it just seems to depend somehow on the size of the enclosed text.

Seems related to #14719.

**Comment by MattijsE** [comment]
I can confirm I have the same issue.
My assuption is that an upgrade of Safari (mac/iOS) has introduced this regression. That's why @josh-byster his deployed version suddenly had the issue.

**Comment by amandaesmith3** [comment]
Thanks for the issue! I've investigated this internally with the team, and we've found that the underlying cause is a Webkit bug that's been resolved as of Safari 15.2.

The problem here is that while `pointer-events: none` is applied to the `ion-button` when it's disabled, Webkit sometimes does not correctly propagate this to the button text (the `.button-inner` element) until there's been a repaint. The result is that clicking the button text specifically can still let click events and other mouse-related behavior through. This only happens under specific conditions:

1. The button was set to `disabled` programmatically.
2. The button text contains only numbers.

I believe that (2) is because letters will trigger the `text-transform: uppercase` CSS applied to `ion-button`, which triggers a repaint and gets `pointer-events: none` propagated correctly. Note that the relation to the text length is a coincidence here; I was able to replicate the bug with any length text, but only with numbers.

The team and I have decided not to update Ionic to account for this, as the only workaround we've found that handles things will most likely have other side effects (see my closed PR linked above), and the bug does not trigger in Safari 15.2. We recommend updating your devices where possible.

I'm going to close this out. Thank you for using Ionic!
